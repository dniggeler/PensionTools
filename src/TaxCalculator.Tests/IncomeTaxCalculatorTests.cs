using System.Threading.Tasks;
using FluentAssertions;
using PensionCoach.Tools.TaxCalculator.Abstractions;
using PensionCoach.Tools.TaxCalculator.Abstractions.Models;
using Snapshooter.Xunit;
using Xunit;

namespace TaxCalculator.Tests
{
    [Trait("Calculator", "Income")]
    public class IncomeTaxCalculatorTests : IClassFixture<TaxCalculatorFixture<IIncomeTaxCalculator>>
    {
        private readonly TaxCalculatorFixture<IIncomeTaxCalculator> _fixture;

        public IncomeTaxCalculatorTests(TaxCalculatorFixture<IIncomeTaxCalculator> fixture)
        {
            _fixture = fixture;
        }

        [Fact(DisplayName = "Income Tax")]
        public async Task ShouldCalculateIncomeTax()
        {
            // given
            var taxPerson = new TaxPerson
            {
                Canton = "ZH",
                CalculationYear = 2018,
                CivilStatus = CivilStatus.Married,
                DenominationType = DenominationType.Married,
                Municipality = "Zürich",
                TaxableIncome = 99995
            };

            // when
            var result = await _fixture.Calculator.CalculateAsync(taxPerson);

            result.IsRight.Should().BeFalse();
            Snapshot.Match(result);
        }
    }
}
