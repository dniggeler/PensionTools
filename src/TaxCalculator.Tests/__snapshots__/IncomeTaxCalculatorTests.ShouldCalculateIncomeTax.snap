[
  {
    "State": "IsRight",
    "Right": {
      "BasisTaxAmount": {
        "DeterminingFactorTaxableAmount": 99900.0,
        "TaxAmount": 4878.0
      },
      "CantonRate": 100.0,
      "MunicipalityRate": 119.0,
      "MunicipalityTaxAmount": 5804.82,
      "CantonTaxAmount": 4878.0,
      "TotalTaxAmount": 10682.82
    },
    "Left": null
  }
]
